hi-ml/testhiml/testhiml/utils/test_common_utils.py

```diff
@@ -19,3 +27,104 @@ def test_set_to_eval_mode() -> None:
+            assert get_all_environment_files(project_root=temp_project_root) == [temp_env_path]
+
+        # Now pass a directory that doesn't have an env file and check that an error is raised
+        with patch("health_azure.paths.git_repo_root_folder", return_value=empty_temp_dir):
+            with pytest.raises(ValueError) as e2:
+                get_all_environment_files(project_root=empty_temp_dir)
+                assert "No Conda environment files were found in the repository" in str(e2)
+
+
+@pytest.mark.fast
+def test_check_conda_environments(temp_env_path: Path) -> None:
+    with patch("health_azure.paths.is_himl_used_from_git_repo", return_value=True):
+        with patch("health_azure.paths.shared_himl_conda_env_file", return_value=temp_env_path):
+            # Calling check_conda_environments on an empty list should do nothing
+            check_conda_environments([])
+
+            # Pass a non-empty list and mock the rturn value of is_conda_file_with_pip_include
+            with patch("health_ml.utils.common_utils.is_conda_file_with_pip_include", return_value=(True, None)):
+                with pytest.raises(ValueError) as e:
+                    check_conda_environments([Path('some_path')])
+                    assert "uses '-r' to reference pip requirements" in str(e)
+
+        # If the file that we pass is the same as the return value of shared_himl_conda_env_file
+        # an error will not be raised
+        with patch("health_azure.paths.shared_himl_conda_env_file", return_value=Path('some_path')):
+            with patch("health_ml.utils.common_utils.is_conda_file_with_pip_include", return_value=(True, None)):
+                check_conda_environments([Path('some_path')])
+
+        # If not is_conda_is_conda_file_with_pip_include, excpect nothing to happen
+        with patch("health_azure.paths.shared_himl_conda_env_file", return_value=temp_env_path):
+            with patch("health_ml.utils.common_utils.is_conda_file_with_pip_include", return_value=(False, None)):
+                check_conda_environments([Path('some_path')])
+
+    # Now check cases where is_himl_used_from_git_repo is False:
+    with patch("health_azure.paths.is_himl_used_from_git_repo", return_value=False):
+        # If not is_conda_is_conda_file_with_pip_include, excpect nothing to happen
+        with patch("health_ml.utils.common_utils.is_conda_file_with_pip_include", return_value=(False, None)):
+            check_conda_environments([])
+            check_conda_environments([Path('some_path')])
+
+        # If is_conda_is_conda_file_with_pip_include=True, expect a ValueError to be raised
+        with patch("health_ml.utils.common_utils.is_conda_file_with_pip_include", return_value=(True, None)):
+            with pytest.raises(ValueError) as e:
+                check_conda_environments([Path('some_path')])
+                assert "uses '-r' to reference pip requirements" in str(e)
```

Comment:
careful, the assert in line 130 needs to be outside of the context manager, it will otherwise not be executed. Double-check that you're not doing that elsewhere too?

Reply:
for example, line 106